src/features/newsSlice.js

```diff
@@ -0,0 +1,33 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+export const newsAPI = createApi({
+  reducerPath: "newsAPI",
+  baseQuery: fetchBaseQuery({ baseUrl: "https://news-proxy.netlify.app/api" }),
+  endpoints: (builder) => ({
+    getNews: builder.query({
+      query: () =>
+        "top-headlines?country=us&pageSize=10&apiKey=9bfc1e38aaee4c038fc09a79c0ea1394",
+      transformResponse: (response) => response.articles,
+    }),
+    getPublisher: builder.query({
+      query: () =>
+        "top-headlines/sources?country=us&apiKey=9bfc1e38aaee4c038fc09a79c0ea1394",
```

Comment:
Application credentials should not be pushed on github, please the API key as environment varible
```suggestion
        "top-headlines/sources?country=us&apiKey=process.env.REACT_APP_API+KEY",
```